src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -154,10 +155,12 @@ class ClusterObserver : IDisposable
                 {
                     try
                     {
-                        if(_decisionsLog.ContainsKey(command.DatabaseName) == false)
-                            _decisionsLog[command.DatabaseName] = new List<CommandBase>();
-                        
-                        _decisionsLog[command.DatabaseName].Add(command);
+                        _decisionsLog.AddOrUpdate(command.DatabaseName,
+                            key => new List<CommandBase> { command },
```

Comment:
You just moved the concurrency issue around.
`List<T>` is not thread safe. Need to see if there is _any_ multi threaded access to it, and there likely is.
At a minimum, you have a thread reading it (the debug endpoint) while it is being modified.